### ISSUE TITLE
New Probe to detect is .gitignore file is present or not

### ISSUE THREAD

**Issue by AayushSaini101** [jenkins-infra/plugin-health-scoring#432]
### Description

New Probe to detect is .gitignore file is present or not

**Comment by alecharp** [comment]
what is the benefit of this probe? I don't really see how it's mandatory to have this file on each plugin repository.

Contributors could have a global ignore file.

**Comment by AayushSaini101** [comment]
@alecharp Thanks , There are multiple advantages of having this file. It helps to increase the code quality of a plugin, andIgnoring unnecessary files and directories using .gitignore helps in **keeping the repository clean and focused on essential code** and assets. This can result in smaller repositories, **faster cloning**, and improved overall performance.

* **Consistent Build and Deployment:**
When deploying applications, having a consistent set of files included and excluded ensures that the build and deployment process is reliable. The presence of a .gitignore file helps in achieving this consistency.

* Prevention of Sensitive Information:

Developers often use .gitignore to exclude files containing sensitive information, such as API keys or configuration files with passwords. Detecting the presence of .gitignore helps ensure that sensitive information is not accidentally committed to the repository.

In short it is very helpful to have,

**Comment by AayushSaini101** [comment]
> @alecharp Thanks , There are multiple advantages of having this file. It helps to increase the code quality of a plugin, andIgnoring unnecessary files and directories using .gitignore helps in **keeping the repository clean and focused on essential code** and assets. This can result in smaller repositories, **faster cloning**, and improved overall performance.
> 
> * **Consistent Build and Deployment:**
>   When deploying applications, having a consistent set of files included and excluded ensures that the build and deployment process is reliable. The presence of a .gitignore file helps in achieving this consistency.
> * Prevention of Sensitive Information:
> 
> Developers often use .gitignore to exclude files containing sensitive information, such as API keys or configuration files with passwords. Detecting the presence of .gitignore helps ensure that sensitive information is not accidentally committed to the repository.
> 
> In short it is very helpful to have,

@alecharp thoughts on this

**Comment by alecharp** [comment]
> There are multiple advantages of having this file. It helps to increase the code quality of a plugin

How so?

> Ignoring unnecessary files and directories using .gitignore helps in keeping the repository clean and focused on essential code and assets

Can be done using a global ignore file.

> This can result in smaller repositories, faster cloning, and improved overall performance

First two items, yes. But I fail to see how for the third.

> When deploying applications, having a consistent set of files included and excluded ensures that the build and deployment process is reliable. The presence of a .gitignore file helps in achieving this consistency.

Not really as the build process will generate the `.class` files required if the dates of the `.java` files are incoherent.

> Developers often use .gitignore to exclude files containing sensitive information, such as API keys or configuration files with passwords. Detecting the presence of .gitignore helps ensure that sensitive information is not accidentally committed to the repository.

If you only rely on the `.gitignore` file to present sensitive information to be published, you are doing it wrong: I can create a `test.txt` and past my password in it and the `.gitignore` file wouldn't be of any help.

We could have this probe added to the project, but I don't see its point. To my knowledge, all repositories have one, new plugin archetype includes one.

And even with one, you can `git add -f target word` in any repository and those folders would be added and tracked for any modifications going forward.
